custom_components/waste_collection_schedule/waste_collection_schedule/source/landkreis_rhoen_grabfeld.py

```diff
@@ -25,43 +25,76 @@
 ]
 
 ICON_MAP = {
-    "Restmüll/Biotonne": "mdi:trash-can",
+    "Restmüll": "mdi:trash-can",
+    "Bio": "mdi:leaf",
     "Gelbe Tonne": "mdi:recycle-variant",
-    "Papiersammlung": "mdi:package-variant",
-    "Problemmüllsammlung": "mdi:biohazard",
+    "Papier": "mdi:package-variant",
+    "Problemmüll": "mdi:biohazard",
 }
 
 
 PARAM_TRANSLATIONS = {
     "de": {
-        "city": "Ort",
-        "district": "Ortsteil",
+        "city": "Stadt, Markt, Gemeinde",
+        "district": "Ort, Ortsteil",
     }
 }
 
-
 class Source:
     def __init__(self, city: str | None = None, district: str | None = None):
         self._city = city
         self._district = district
 
     def fetch(self):
         r = requests.get(
-            API_URL, params={"stadt": self._city, "ortsteil": self._district}
+            API_URL, params={"module_division_uuid": "fde08d95-111b-11ef-bbd4-b2fd53c2005a"}
         )
 
         r.raise_for_status()
 
+        city_id = None
+        area_id = None
+
+        json = r.json()
+        config = json["mdiv"]["config"]
+
+        if self._city is not None:
+            # determine city id
+            for city in config["cities"]:
+                if city["name"] == self._city:
+                    city_id = int(city["id"])
+                    break
+            if city_id is None:
+                raise Exception(f"'{self._city}' is not a valid city")
+
+        if self._district is not None:
+            # determine district id
+            for area in config["areas"]:
+                if area["name"] == self._district and int(area["city_id"]) == city_id:
+                    area_id = int(area["id"])
+            if area_id is None:
+                raise Exception(f"'{self._area}' is not a valid district")
```

Comment:
You could raise `SourceArgumentNotFoundWithSuggestions` here (`from waste_collection_schedule.exceptions import SourceArgumentNotFoundWithSuggestions`) to provide a better experience configuring through the GUI

Reply:
Thanks for the hint. I was not yet aware of this feature.
I added it as suggested.